src/gps/dateTime.c

```diff
@@ -133,3 +133,118 @@ tiny_millis_t getUptime() {
 int getUptimeAsInt() {
    return (int) getUptime();
 }
+
+void getDateTimeFromEpochMillis(DateTime *dateTime, millis_t millis) {
```

Comment:
> This isn't complicated or tricky, and arguably if other people looking at this code have no idea that this is a thing, they probably aren't the people you want contributing to your codebase.

Such a comment on par with Linus.  Love it.

> This hardly feels like premature optimization, it is a coding style preference.

Fair point.

> What if this function could fail? How would we return an error condition?

This method doesn't do that so its not a concern of this particular discussion.  It would be a different case should that functionality be needed.  That in itself is a reason to follow the design in question.  But that is not the case here.


Ok.  This horse is dead enough.  This is obviously a difference in design mentalities that will not be resolved in a code review (if it need be).  Lets get something from all this discussion.  Do we want to pass all structures by reference to keep stack space small as a general practice?  This will at least keep things consistent going forward.